GitUI/BranchTreePanel/RepoObjectsTree.cs

```diff
@@ -163,34 +164,49 @@ public async Task ReloadAsync()
                 treeMain.BeginUpdate();
                 _rootNodes.ForEach(t => t.IgnoreSelectionChangedEvent = true);
                 var tasks = _rootNodes.Select(r => r.ReloadAsync(token)).ToArray();
-                await Task.WhenAll(tasks).ConfigureAwait(false);
+                await Task.WhenAll(tasks).ConfigureAwait(true);
             }
             finally
             {
-                await this.SwitchToMainThreadAsync();
+                ThreadHelper.AssertOnUIThread();
+
                 if (!token.IsCancellationRequested)
                 {
                     Enabled = true;
+                }
+
+                // Need to end the update for the selected node to scroll into view below under certain conditions
```

Comment:
The reason I moved EndUpdate higher up here, before we set the selected node, is that there's a weird TreeView bug that occurs when we first populate with more nodes, then set a selected node, while in a Begin/EndUpdate pair, in which case the node will not be scrolled into view. By ending the update earlier, and then setting the selected node, this always works at the cost of the user potentially seeing the result of the scroll occur. I think it's an acceptable compromise given that it will always correctly ensure the selected node is visible.